src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -1008,6 +1008,63 @@ type internal TypeCheckInfo
             |> List.prependIfSome (CreateCompletionItemForSuggestedPatternName caseIdPos fieldName))
         |> Option.defaultValue completions
 
+    /// Gets all methods that a type can, but has not yet, overriden. Object methods are not included.
+    let GetOverridableMethods pos enclosingTypeNameRange inheritTypeNameRange =
+        let methodNotAlreadyOverridden (alreadyOverriddenMethods: MethInfo list) (candidate: MethInfo) g =
+            alreadyOverriddenMethods
+            |> List.forall (fun x ->
+                if candidate.DisplayName <> x.DisplayName then
+                    true
+                else
+                    match x, candidate with
+                    | FSMeth (valRef = vref1), FSMeth (valRef = vref2) ->
+                        match stripTyEqns g vref1.Type, stripTyEqns g vref2.Type with
+                        | TType_fun (rangeType = ty1), TType_fun (rangeType = ty2) -> not (typeEquiv g ty1 ty2)
+                        | _ -> not (typeEquiv g vref1.Type vref2.Type)
+                    | _ -> false)
```

Comment:
This is really half-assed, and doesn't work for IL classes, but I could not find a simple way for matching methods by signature. It's a complex topic, so there might not be one.